ha/core/event_analyzer/parser/parser.py

```diff
@@ -188,9 +188,10 @@ def parse_event(self, msg: str) -> HealthEvent:
             resource_id = cluster_resource_alert[EventAttr.EVENT_PAYLOAD.value][HealthAttr.RESOURCE_ID.value]
             event_type = cluster_resource_alert[EventAttr.EVENT_PAYLOAD.value][HealthAttr.RESOURCE_STATUS.value]
             specific_info = cluster_resource_alert[EventAttr.EVENT_PAYLOAD.value][HealthAttr.SPECIFIC_INFO.value]
-            if specific_info and specific_info["generation_id"] and source == HEALTH_EVENT_SOURCES.MONITOR.value:
-                generation_id = cluster_resource_alert[EventAttr.EVENT_PAYLOAD.value][HealthAttr.SPECIFIC_INFO.value]["generation_id"]
-                specific_info = {"generation_id": generation_id, "pod_restart": 0}
+            if resource_type == CLUSTER_ELEMENTS.NODE.value:
+                if specific_info and specific_info["generation_id"] and source == HEALTH_EVENT_SOURCES.MONITOR.value:
```

Comment:
What is the impact of removing the check source == HEALTH_EVENT_SOURCES.MONITOR.value: ?
IF there is no impact on current behavior of fault_tolerance, please remove that check. 
If some impact is identified,  add a TODO here
# TODO remove the check source == HEALTH_EVENT_SOURCES.MONITOR.value and validate the corner cases 
and  create a jira for resolving this TODO

Reply:
No impact after removing this condition, I removed this condition now.